File: python/objects/critter.py
""" Critter.py
    Basic critter class
    includes a constructor
"""
print "In critter.py, namespace is %s" % __name__
 
class Critter(object):
  def __init__(self):
    object.__init__(self)
    self.name = "Anonymous"
    
  def sayHi(self):
    print "Hi, my name is %s!" % self.name
    

def main():  
  c = Critter()
  c.name = "George"
  c.sayHi()
  
main()

File: inheritance/Critter.cpp
#include <string>
#include <iostream>

#include "Critter.h"

using namespace std;

//Constructor and greet are defined for Critter
Critter::Critter(string name){
  Critter::name = name;
};

void Critter::greet(){
  cout << "My name is " << name << "." << endl;
} // end greet

int main(){

  cout << "main class: ";
  Critter c;
  c.greet();

  return 0;
}

